### ISSUE TITLE
Local media files don't work on aldryn-django 2.2.12.2a3 when CH Azure region is used

### ISSUE THREAD

**Issue by viktor-yunenko** [what-digital/divio#2]
I pulled media from the project https://control.divio.com/control/6276/edit/82867/ using `divio project pull media test` command.

Images that I'm looking at weren't visible in the admin:

![image](https://user-images.githubusercontent.com/36416236/85134356-04b9e280-b245-11ea-87a7-4c1977616080.png)

It looks the images always aren't shown when {% thumbnail %} tag is in use - it renders something as `<img src="">`, but adding:
```
MEDIA_ROOT = os.path.join(DATA_ROOT, 'media/public-media')
```
to `settings.py` at least fixes direct images invocations eg as `<img src="{{ object.hero_image.url }}">`

Also the files that I'm uploading locally are added to the root of my git project, instead of `data/media/public-media/`.

<details>

![image](https://user-images.githubusercontent.com/36416236/86144879-e97d8b80-bafe-11ea-9867-55ec4a0f7314.png)

</details>

Basically in `data/media/public-media` we can see the unpacked dump from Azure, and `landspace.jpg` is a file that I tried to upload on my local instance

Snapshot of media settings from docker

```
'DEFAULT_FILE_STORAGE': 'aldryn_django.storage.DefaultStorage'
'MEDIA_ROOT': '/app/data/media'
'MEDIA_URL': '/media/',
'DEFAULT_STORAGE_DSN': 'file://?url=%2Fmedia%2F',
```

**Comment by viktor-yunenko** [comment]
`public-media` seems to be an important part of the site, especially on the divio server. Locally though django ignores it and still tries to save data to `data/media/`?

@yvladislav , does that correspond to what you have seen on your current project?

**Comment by viktor-yunenko** [comment]
I can see that it happens because locally the setup (aldryn-django probably?) sets the media path to `/media/` while it must be `/media/public-media/`

**Comment by viktor-yunenko** [comment]
The following doesn't work
- http://127.0.0.1:8000/media/filer_public/ab/b6/abb60f1c-3c0a-4809-a948-ce439928cc04/srk_visual.jpg/

While with media-public it does
- http://127.0.0.1:8000/media/public-media/filer_public/ab/b6/abb60f1c-3c0a-4809-a948-ce439928cc04/srk_visual.jpg/

**Comment by viktor-yunenko** [comment]
I fixed it by changing 

        settings['MEDIA_ROOT'] = env('MEDIA_ROOT', os.path.join(settings['DATA_ROOT'], 'media'))

to 

        settings['MEDIA_ROOT'] = env('MEDIA_ROOT', os.path.join(settings['DATA_ROOT'], 'media/public-media'))

**Comment by viktor-yunenko** [comment]
Hi @FinalAngel, is this something that we should add to aldryn-django? The non-working media on local instances makes the frontend development quite challenging at the moment :)

I guess it might be fixed in the latest version of aldryn-django, but Jonathan said that this project we must keep on 2.2.12.2a3 version, since the others don't work with Azure.

**Comment by viktor-yunenko** [comment]
Although now I'm confused, it works for some media files but not for others. And of course divio bot rolls back those changes :(

**Comment by viktor-yunenko** [comment]
I wasn't able to add the MEDIA_ROOT update to settings.py, I guess that's one of those changes that must be added before `aldryn_addons.settings.load(locals())` (eg as db settings?) - otherwise they have no effect. Hence I tried to add them to the addons/ file, but divio bot resets them right after.

**Comment by viktor-yunenko** [comment]
Reply from Dennis:

> Also, do you happen to know why it wants to save files to public-media? I am not familiar with that folder. 

No idea :)

I'm pretty sure it happened after the migration to CH AZ region, Jonathan was mentioning some issues with Azure I think?

**Comment by viktor-yunenko** [comment]
I'm probably mistaken about settings.py - it looks the images always aren't shown when `{% thumbnail %}` tag is in use, but adding:
```
MEDIA_ROOT = os.path.join(DATA_ROOT, 'media/public-media')
```
to `settings.py` at least fixes direct images invocations eg as `<img src="{{ object.hero_image.url }}">`

**Comment by viktor-yunenko** [comment]
Also the files that I'm uploading locally are added to the root of my git project, instead of data/media/public-media/.

![image](https://user-images.githubusercontent.com/36416236/86144757-be933780-bafe-11ea-915b-84bbfc14ea16.png)

**Comment by viktor-yunenko** [comment]
I can confirm that rolling back a working project to aldryn-django 2.2.12.2a3 (from aldryn-django 2.2.13.1) replicates the issue.

**Comment by viktor-yunenko** [comment]
So the following
```html
<li>url: {{ project.hero_image.url }}</li>
<li>thumbnail: {% thumbnail project.hero_image 361x245 %}</li>
```
renders as
```html
<li>url: /media/filer_public/d3/44/d344ce48-8fe8-4f36-8388-5fe2f820a5d3/photo-1478728073286-db190d3d8ce6.jpeg</li>
<li>thumbnail: </li>
```

I believe django-filer on divio uses some magic addition for S3 usage, might it be involved?

We once tried to use a normal django-filer from pypi and all media images stopped being visible.

**Comment by viktor-yunenko** [comment]
Ok so I can see the bug in aldryn-django `aldryn_config.py`, this is how it looks like right now:

```python
        storage_dsn = env(DEFAULT_STORAGE_KEY, )
        if not storage_dsn:
            dsn = furl.furl()
            dsn.scheme = 'file'
            # MEDIA_ROOT is used before it gets defined below, ie:
            # settings['MEDIA_ROOT'] == ''
            dsn.path = settings['MEDIA_ROOT']
            dsn.args.set("url", env('MEDIA_URL', '/media/'))
            storage_dsn = six.text_type(dsn)
        settings[DEFAULT_STORAGE_KEY] = storage_dsn
        # [...]
        settings['MEDIA_ROOT'] = env('MEDIA_ROOT', os.path.join(settings['DATA_ROOT'], 'media'))
```
The last line defines the MEDIA_ROOT, *before* it gets set on the DSN. Instead the MEDIA_ROOT should be defined before the DSN block:
```python
        settings['MEDIA_ROOT'] = env('MEDIA_ROOT', os.path.join(settings['DATA_ROOT'], 'media/public-media'))

        storage_dsn = env(DEFAULT_STORAGE_KEY, )
        if not storage_dsn:
            dsn = furl.furl()
            dsn.scheme = 'file'
            dsn.path = settings['MEDIA_ROOT']
            dsn.args.set("url", env('MEDIA_URL', '/media/'))
            storage_dsn = six.text_type(dsn)
        settings[DEFAULT_STORAGE_KEY] = storage_dsn
```
After that filer thumbnails work normally.

**Comment by viktor-yunenko** [comment]
Basically adding the following to settings.py works
```python
if DIVIO_ENV == DivioEnv.LOCAL:
    MEDIA_ROOT = os.path.join(DATA_ROOT, 'media/public-media')
    dsn = furl.furl()
    dsn.scheme = 'file'
    dsn.path = MEDIA_ROOT
    dsn.args.set("url", env.get('MEDIA_URL', '/media/'))
    storage_dsn = str(dsn)
    from aldryn_django.storage import DEFAULT_STORAGE_KEY
    locals()[DEFAULT_STORAGE_KEY] = storage_dsn
```

**Comment by macolo** [comment]
azure is now supported by the latest aldryn-django version according to divio support. Needs to be tested.
